src/main/java/com/mathworks/ci/tools/MatlabInstaller.java

```diff
@@ -0,0 +1,308 @@
+package com.mathworks.ci.tools;
+/**
+ * Copyright 2024, The MathWorks, Inc.
+ *
+ */
+
+
+import com.mathworks.ci.MatlabInstallation;
+import com.mathworks.ci.Message;
+import com.mathworks.ci.utilities.GetSystemProperties;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.Launcher.ProcStarter;
+
+import hudson.model.Node;
+import hudson.model.TaskListener;
+import hudson.tools.DownloadFromUrlInstaller;
+import hudson.tools.ToolInstallation;
+import hudson.tools.ToolInstallerDescriptor;
+
+import hudson.util.ArgumentListBuilder;
+import hudson.util.FormValidation;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URL;
+
+import java.nio.charset.StandardCharsets;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Locale;
+
+import java.util.Set;
+import jenkins.model.Jenkins;
+import org.apache.commons.io.IOUtils;
+
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+import org.kohsuke.stapler.QueryParameter;
+import org.kohsuke.stapler.verb.POST;
+
+public class MatlabInstaller extends DownloadFromUrlInstaller {
+
+    private String version;
+    private String products;
+
+    @DataBoundConstructor
+    public MatlabInstaller (String id) {
+        super (id);
+    }
+
+    public String getVersion () {
+        return this.version;
+    }
+
+    @DataBoundSetter
+    public void setVersion (String version) {
+        this.version = version;
+    }
+
+    public String getProducts () {
+        return this.products;
+    }
+
+    @DataBoundSetter
+    public void setProducts (String products) {
+        this.products = products;
+    }
+
+    @Override
+    public FilePath performInstallation (ToolInstallation tool, Node node, TaskListener log)
+        throws IOException, InterruptedException {
+        FilePath supportingExecutable = preferredLocation (tool, node);
+        FilePath expectedPath = new FilePath (supportingExecutable, this.getVersion ());
+        MatlabInstallable installable;
+        try {
+            installable = (MatlabInstallable) getInstallable (node);
+        } catch (Exception e) {
+            throw new InstallationFailedException (e.getMessage ());
+        }
+
+        getFreshCopyOfExecutables (installable, supportingExecutable);
+        makeDir (expectedPath);
+
+        FilePath versionInfo = new FilePath (expectedPath,"VersionInfo.xml");
+        FilePath installedProducts = new FilePath (expectedPath,"installed_matlab_product_list.txt");
+        if (versionInfo.exists () && isSameProduct(installedProducts)) {
+            return expectedPath;
+        } else {
+            int result = installUsingMpm (node, expectedPath, log, installedProducts);
+            if (result == 0) {
+                log.getLogger ().println (
```

Comment:
Installation of version ... using `mpm` completed successfully!

Reply:
Done , Keeping MATLAB prefix as it is as it will help us differentiate amounts other product installations.